### PR TITLE
[Fix] Fix counter mapping bug

### PR DESCRIPTION
**Pull request by gaoyang07**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: tests/test_models/test_task_modules/test_estimators/test_flops_params.py [hunk 1]
```diff
@@ -127,6 +128,15 @@ def test_estimate(self) -> None:
         self.assertGreater(flops_count, 0)
```
Comment (sunnyxiaohu) [inline tests/test_models/test_task_modules/test_estimators/test_flops_params.py:128]: add more strict check constraints for `::test_estimate` UTs, not just greater than 0.

File: mmrazor/models/task_modules/estimators/counters/flops_params_counter.py [hunk 2]
```diff
@@ -409,6 +411,12 @@ def add_flops_params_counter_hook_function(module):
 
             else:
                 counter_type = get_counter_type(module)
+                if counter_type not in TASK_UTILS._module_dict.keys():
+                    old_counter_type = counter_type
```
Comment (sunnyxiaohu) [inline mmrazor/models/task_modules/estimators/counters/flops_params_counter.py:415]: Move the logic into `get_counter_type` and then you can also refactor `is_supported_instance`.

File: mmrazor/models/task_modules/estimators/counters/flops_params_counter.py [hunk 3]
```diff
@@ -503,9 +511,13 @@ def get_counter_type(module):
 
 
 def is_supported_instance(module):
-    """Judge whether the module is in TASK_UTILS registry or not."""
+    """Judge whether the module can be countered or not."""
     if get_counter_type(module) in TASK_UTILS._module_dict.keys():
         return True
+    else:
+        for op in get_modules_list():
+            if issubclass(module.__class__.__base__, op):
```
Comment (sunnyxiaohu) [inline mmrazor/models/task_modules/estimators/counters/flops_params_counter.py:519]: multiple inheritances should also be handled.

File: mmrazor/models/task_modules/estimators/counters/flops_params_counter.py [hunk 4]
```diff
@@ -409,6 +411,12 @@ def add_flops_params_counter_hook_function(module):
 
             else:
                 counter_type = get_counter_type(module)
+                if counter_type not in TASK_UTILS._module_dict.keys():
+                    old_counter_type = counter_type
+                    counter_type = \
+                        module.__class__.__base__.__name__ + 'Counter'
+                    warnings.warn(f'`{old_counter_type}` not in '
```
Comment (sunnyxiaohu) [inline mmrazor/models/task_modules/estimators/counters/flops_params_counter.py:418]: I suggest using `logger.warning` from mmengine.

File: tests/test_models/test_task_modules/test_estimators/test_flops_params.py (hunk 1 above)
Comment (gaoyang07) [reply tests/test_models/test_task_modules/test_estimators/test_flops_params.py]: Done.

File: mmrazor/models/task_modules/estimators/counters/flops_params_counter.py (hunk 2 above)
Comment (gaoyang07) [reply mmrazor/models/task_modules/estimators/counters/flops_params_counter.py]: Done.

File: mmrazor/models/task_modules/estimators/counters/flops_params_counter.py (hunk 3 above)
Comment (gaoyang07) [reply mmrazor/models/task_modules/estimators/counters/flops_params_counter.py]: Use `module.__class__.mro()` to handle multiple inheritances, done.

File: mmrazor/models/task_modules/estimators/counters/flops_params_counter.py (hunk 4 above)
Comment (gaoyang07) [reply mmrazor/models/task_modules/estimators/counters/flops_params_counter.py]: Done.

**Review by sunnyxiaohu — APPROVED** [review]

**Pull request merged**